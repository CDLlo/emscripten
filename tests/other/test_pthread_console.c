#include <stdio.h>
#include <emscripten/em_asm.h>

void out(const char* msg) {
  EM_ASM({ console.log(UTF8ToString($0)); }, msg);
}

// Test that stdout/printf and console.log message are interleaved as expected
// and all arrive at the console.
// See https://github.com/emscripten-core/emscripten/issues/14804
int main() {
  printf("printf 1\n");
  out("console.log 1");
  printf("printf 2\n");
  out("console.log 2");
  printf("printf 3\n");
  out("console.log 3");
  printf("done\n");
  return 0;
}
