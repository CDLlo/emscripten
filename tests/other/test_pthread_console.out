printf 1
console.log 1
printf 2
console.log 2
printf 3
console.log 3
done
